src/commands/deleteVirtualMachine/deleteAllResources.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ResourceManagementClient } from "@azure/arm-resources";
+import { ISubscriptionContext } from "vscode-azureextensionui";
+import { localize } from "../../localize";
+import { createResourceClient } from "../../utils/azureClients";
+import { networkInterfaceLabel, ResourceDeleteError, ResourceToDelete, virtualMachineLabel, virtualNetworkLabel } from "./deleteConstants";
+import { deleteWithOutput } from "./deleteWithOutput";
+
+export async function deleteAllResources(context: ISubscriptionContext, resourceGroupName: string, resourcesToDelete: ResourceToDelete[]): Promise<ResourceDeleteError[]> {
+    const errors: ResourceDeleteError[] = [];
+
+    // virtual machines have to be deleted before a lot of other resources so do it first
+    const virtualMachineIndex: number = resourcesToDelete.findIndex(r => { return r.resourceType === virtualMachineLabel; });
+    if (virtualMachineIndex >= 0) {
+        await deleteWithOutput(resourcesToDelete.splice(virtualMachineIndex, 1)[0], errors);
+    }
+
+    // network interfaces have to be delete before public IP and virtual networks
+    const networkInterfaceIndex: number = resourcesToDelete.findIndex(r => { return r.resourceType === networkInterfaceLabel; });
+    if (networkInterfaceIndex >= 0) {
+        await deleteWithOutput(resourcesToDelete.splice(networkInterfaceIndex, 1)[0], errors);
+    }
+
+    // virtual networks have to be deleted before nsg but after network interface
+    const virtualNetworkIndex: number = resourcesToDelete.findIndex(r => { return r.resourceType === virtualNetworkLabel; });
+    if (virtualNetworkIndex >= 0) {
+        await deleteWithOutput(resourcesToDelete.splice(virtualNetworkIndex, 1)[0], errors);
+    }
+
+    await Promise.all(resourcesToDelete.map(async r => {
+        await deleteWithOutput(r, errors);
+    }));
```

Comment:
This code feels a bit too duplicate-y to me and I don't like how you needed to make the comment about "the message needs to constructed here" because of the use of splice. How about something like this:

```typescript
    // virtual machines have to be deleted before a lot of other resources so do it first
    // network interfaces have to be delete before public IP and virtual networks
    // virtual networks have to be deleted before nsg but after network interface
    // the rest can be deleted in parallel
    const orderedLabels: string[] = [virtualMachineLabel, networkInterfaceLabel, virtualNetworkLabel];
    const serialResources: ResourceToDelete[] = [];
    const parallelResources: ResourceToDelete[] = [];
    for (const resource of resourcesToDelete) {
        orderedLabels.includes(resource.resourceType) ? serialResources.push(resource) : parallelResources.push(resource);
    }

    serialResources.sort((a, b) => orderedLabels.indexOf(a.resourceType) - orderedLabels.indexOf(b.resourceType));
    for (const resource of serialResources) {
        await deleteWithOutput(resource, errors);
    }

    await Promise.all(parallelResources.map(async r => await deleteWithOutput(r, errors)));
```